src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const anwser = [];
+  let copyOfState = { ...state };
+
+  for (const actionObjects of actions) {
+    const copyBeforePushing = { ...copyOfState };
+
+    switch (actionObjects.type) {
+      case 'addProperties':
+        for (const data in actionObjects.extraData) {
+          copyBeforePushing[data] = actionObjects.extraData[data];
+        }
+
+        anwser.push(copyBeforePushing);
+        copyOfState = copyBeforePushing;
+        break;
+      case 'removeProperties':
+        for (const aKeyToRemove of actionObjects.keysToRemove) {
+          if (aKeyToRemove in copyOfState) {
+            delete copyBeforePushing[aKeyToRemove];
+          }
+        }
+
+        anwser.push(copyBeforePushing);
+        copyOfState = copyBeforePushing;
+        break; ;
+      case 'clear':
+        for (const stateKey in copyOfState) {
+          delete copyBeforePushing[stateKey];
+        }
+        anwser.push(copyBeforePushing);
+        copyOfState = copyBeforePushing;
+        break;
+    }
```

Comment:
```suggestion
    }
    
    answer.push({ ...copyOfState })
```
push делать один раз после switch, все равно у тебя только один из case'ов выполняется на каждом шаге

Reply:
Спасибо, уже понял :)